addons/crm/models/crm_lead.py

```diff
@@ -679,6 +679,12 @@ def search(self, args, offset=0, limit=None, order=None, count=False):
         )
         return self.browse(my_lead_ids_keep) + other_lead_res
 
+    @api.constrains('stage_id')
+    def check_stage_id(self):
+        for lead in self:
```

Comment:
could be list-comprehension-ed, like `` if any(not lead.active for lead in self): ... ``